src/Symfony/Component/Serializer/Deserialize/Template/TemplateGeneratorInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\Template;
+
+use Symfony\Component\Serializer\Deserialize\Config\DeserializeConfig;
+use Symfony\Component\Serializer\Deserialize\DataModel\DataModelNodeInterface;
+use Symfony\Component\Serializer\Php\PhpNodeInterface;
+
+/**
+ * Generates a deserialization template PHP syntax tree based on a given data model.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+interface TemplateGeneratorInterface
```

Comment:
Do you have an use case for this extension point ? I assume this may be needed for api platform ?

Reply:
Actually, if you are speaking about **deserialization**, no indeed I don't see any reason to have that interface accessible, we can turn it to internal.

But, if you are talking about **serialization**, this interface is the only way to implement a fully custom template generator, which allows serializing a new format efficiently (without using the `NormalizerEncoderTemplateGenerator`). That being said, I don't think that regular developers will leverage that extension point. But it can be the case for bundles, I guess.

Reply:
Again, we should avoid to guess the usage of a “feature” until absolutely necessary 🙏🏻